### DISCUSSION TITLE
Keeping older micronaut version while using a library with newer version

### DISCUSSION THREAD

**Discussion by martisaw** [micronaut-projects/micronaut-core#5495] [Q&A]
I have a project with an older Micronaut version (`2.4.4`). I am using Gradle for dependency management and want to use `micronaut-camunda-bpm` in my project. This project uses the micronaut library plugin and already uses Micronaut `2.5.4`. When I add this dependency to my project, Gradle updates all my current `2.4.4` dependencies to `2.5.4`. 

Is there an easy solution to downgrade the dependencies of micronaut-camunda-bpm to `2.4.4` so that it matches my project? 

I already tried to set strict versions on the dependencies of my project, but I ended with some dependencies on `2.4.4` and the rest on `2.5.4`.

**Comment by ilopmar** [comment]
Not really answering your question but is there a reason why you can't upgrade your project from 2.4.4 to 2.5.4? They upgrade should work without issues (probably)

**Reply by martisaw** [reply]
yeah I could upgrade, but I want to check the downgrade compatibility of the `micronaut-camunda-bpm` project. To see if it also works with an older micronaut project. Therefore I want to match the micronaut version with my `2.4.4` project.

**Reply by martisaw** [reply]
Hey @ilopmar, Do you have any ideas for this? Especially with the upcoming Micronaut 3.x.x this will be an use case for our project [micronaut-camunda-bpm](https://github.com/NovatecConsulting/micronaut-camunda-bpm).

**Reply by ilopmar** [reply]
You can force Gradle to use a specific version of a dependency. Doing that for Micronaut should be enough (I think)

**Comment by jameskleeh** [comment]
In general there is no guarantees that a module compiled against Micronaut 2.5 will work with 2.4. There may have been a feature introduced in 2.5 that is being used and thus is incompatible with 2.4. Library authors should make it clear what version of Micronaut is supported and if they desire to support older versions of Micronaut they should only depend on those older versions or create multiple release lines of their modules.
